### ISSUE TITLE
HTML Comments Not Colour Coded when there is a <?php> in the script tag

### ISSUE THREAD

**Issue by ChristianTheMinery** [Microsoft/vscode#28354]
- VSCode Version: 1.13.0
- OS Version: macOS Sierra 10.12.4

Since updating to version 1.13.0 of VSCode, HTML comments are no longer colour-coded grey, but instead simply appear in plain white.

![screen shot 2017-06-09 at 10 49 34 am](https://user-images.githubusercontent.com/23169829/26980853-5bf30948-4d01-11e7-8915-4f35eaf842ae.png)

**Comment by serge-io** [comment]
Hey Christian, 
I tried to reproduce in my environment and I cannot reproduce.

- VSCode: 1.13.0
- OSVersion: macOS Sierra 10.12.5 

![comment](https://user-images.githubusercontent.com/26391742/27003645-d8375e06-4dfa-11e7-8cb0-f3f4b9d292d9.png)

**Comment by ramya-rao-a** [comment]
Works for me too

@ChristianTheMinery What theme are you using? Can you try other themes and see if you get the same issue?

**Comment by ChristianTheMinery** [comment]
I believe I've identified the issue - the problem is that farther up the file, there's a PHP tag nested inside a script tag:

```
<script>
    ...
        <?php
            foreach($actID AS $act) {
                echo 'divNames.push(\'[nid='.$act.']\');';
             }
        ?>
    ...
</script>
```

And _this_ is what seems to be breaking the 'colour parsing'. It didn't before, but it does now.

**Comment by ramya-rao-a** [comment]
Thanks @ChristianTheMinery 

There was a change to the php grammar this time, so may be that caused this.

cc @roblourens

**Comment by roblourens** [comment]
I do see this, but it doesn't seem like a regression vs 1.12. You're saying that something like this worked in 1.12?

1.12.2:
![image](https://user-images.githubusercontent.com/323878/27105697-60eedd4a-5045-11e7-92d4-e0c58e51005b.png)

**Comment by ChristianTheMinery** [comment]
Yes, it definitely did. I was actually working on this exact file when I updated to 1.13. Before the update, the syntax highlighting worked fine, after, it didn't. Very odd that it doesn't seem to have the same behaviour in your 1.12, though.

**Comment by roblourens** [comment]
Maybe you were using 1.11? It looks correct in 1.11.2:

![image](https://user-images.githubusercontent.com/323878/27144818-25059424-50e8-11e7-836f-ba51968f8929.png)

**Comment by roblourens** [comment]
@aeschli It looks like we supported multiple levels of nesting in 1.11 but not 1.12 or 1.13. The screenshot above includes HTML, JS, and PHP scopes. Do you know what could have happened? An issue with one of the grammars or something else?

**Comment by ChristianTheMinery** [comment]
I could have been on 1.11. I'll admit I'm not the _best_ at keeping my software up to date!

**Comment by aeschli** [comment]
@roblourens  I found the problem.
VSCode uses the HTML grammar from [textmate/html.tmbundle](https://github.com/textmate/html.tmbundle) which has recently moved away from the scope `source.js.embedded.html`. The PHP grammar is from Atom and still uses the scope.

The PHP injections rule that textmate now uses is `L:text.html.php source.js`.
 See [here](https://github.com/textmate/php.tmbundle/commit/1d91862a3d118827aa4c0fb57672625a704410c0).

I patched our instance of the PHP grammar for now.

I'm not sure if is an option for the atom/php grammar to make the same change. https://github.com/atom/language-php/issues/201 has the discussion.
